force-app/main/default/classes/ServiceDeliveryController_TEST.cls

```diff
@@ -77,4 +77,73 @@ public with sharing class ServiceDeliveryController_TEST {
             contactId
         );
     }
+
+    @IsTest
+    private static void testGetServicesByProgramEngagementIds() {
+        Id programEngagementId = TestUtil.mockId(ProgramEngagement__c.SObjectType);
+
+        List<Object> expected = new List<Object>();
+
+        serviceStub.withReturnValue(
+            'getServicesByProgramEngagementId',
+            Id.class,
+            expected
+        );
+
+        Test.startTest();
+        ServiceDeliveryController.service = (ServiceService) serviceStub.createMock();
+
+        final List<Object> actual = ServiceDeliveryController.getServicesByProgramEngagementId(
+            programEngagementId
+        );
+
+        Test.stopTest();
+
+        System.assertEquals(
+            expected.size(),
+            actual.size(),
+            'Expected the size of both expected and actual are the same'
+        );
+
+        serviceStub.assertCalledWith(
+            'getServicesByProgramEngagementId',
+            Id.class,
+            programEngagementId
+        );
+    }
+
+    @IsTest
+    private static void testGetServicesByProgramEngagementIdsException() {
+        Id programEngagementId = TestUtil.mockId(ProgramEngagement__c.SObjectType);
+
+        serviceStub.withThrowException('getServicesByProgramEngagementId', Id.class);
+
+        Test.startTest();
+        ServiceDeliveryController.service = (ServiceService) serviceStub.createMock();
+
+        final List<Object> actual;
```

Comment:
We could do a System assert on this as well and verify it is still null after an exception is thrown.